internal/websocket/options.go

```diff
@@ -218,16 +178,38 @@ func SendTimeout(d time.Duration) Option {
 		})
 }
 
-// ConnectTimeout sets the timeout for the WS connection.  If this is not set,
-// the default is 30 seconds.
-func ConnectTimeout(d time.Duration) Option {
+// HTTPClient is the configuration for the HTTP client used for connection attempts.
+func HTTPClient(c arrangehttp.ClientConfig) Option {
```

Comment:
Instead of passing the `arrangehttp.ClientConfig` pass the `*http.Client` value here.  This is consistent with other similar interfaces & decouples configuration from use a bit more.  Also I don't think we need to boundary check the client since we're being told what someone wants & that will remove some code as well.